src/mlpack/methods/ne/genome.hpp

```diff
@@ -0,0 +1,304 @@
+/**
+ * @file genome.hpp
+ * @author Bang Liu
+ *
+ * Definition of the Genome class.
+ */
+#ifndef MLPACK_METHODS_NE_GENOME_HPP
+#define MLPACK_METHODS_NE_GENOME_HPP
+
+#include <cassert>
+#include <map>
+
+#include <mlpack/core.hpp>
+
+#include "gene.hpp"
+#include "utils.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class defines a genome.
+ # A genome is consist of a group of genes.
+ */
+class Genome {
+ public:
+  // Default constructor.
+  Genome() {}
+  
+  // Parametric constructor.
+  Genome(unsigned int id,
+  	     std::vector<NeuronGene> neuronGenes,
```

Comment:
I think it would be a good idea to use a reference here to avoid copying the object:

``
Genome(unsigned int id, const std::vector<NeuronGene>& neuronGenes, ...)
``